src/compatTest/kotlin/io/github/gradlenexus/publishplugin/MethodScopeWiremockResolver.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import org.junit.jupiter.api.extension.AfterEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.extension.ParameterContext
+import org.junit.jupiter.api.extension.ParameterResolver
+import ru.lanwen.wiremock.ext.WiremockResolver
+import java.lang.IllegalArgumentException
+
+/**
+ * Composes a [WiremockResolver] and uses that by default. But, if a parameter is annotated
+ * with [MethodScopeWiremockResolver] it creates a new instance of the [WiremockResolver] extension and
+ * manages its lifecycle in the scope of that [ExtensionContext]
+ */
+class MethodScopeWiremockResolver(
+    private val inner: WiremockResolver = WiremockResolver(),
+    private val localResolvers: MutableMap<String, WiremockResolver> = mutableMapOf()
```

Comment:
Extensions should not have mutable state but store it in the `ExtensionContext.Store` but we/I can address that after this PR is merged.

Reply:
Ah, definitely agreed! This was easy enough for me to fix now 😃  - I'm a fan of Extension API so always enjoy the opportunity to work with it more